model/testing/src/icon4py/model/testing/helpers.py

```diff
@@ -52,6 +57,10 @@ def is_roundtrip(backend) -> bool:
     return backend.name == "roundtrip" if backend else False
 
 
+def extract_backend_name(backend) -> str:
+    return backend.name if hasattr(backend, "name") else "embedded"
```

Comment:
I would go with 
```suggestion
def extract_backend_name(backend) -> str:
    return "embedded" if backend is None else backend.name
```
because this gives an error in non-intended use cases. E.g. `extract_backend_name(1)`